pipeline/matching/core/engine.py

```diff
@@ -19,6 +25,9 @@ class UserNotMatchedException(Exception):
 
 
 class MatchingEngine:
+
+    analytics: MatchingInternalAnalytics  # analytics are saved to this variable w/out affecting the matching engine output
```

Comment:
Move this into the __init__ method, it should be an instance variable instead of a class variable. More about the differences between those two in this article: https://pynative.com/python-class-variables

Also initialize the object.

Reply:
This has me confused because I was looking over the issue you opened (#224) and in it mentions this: "Create a **class variable** in MatchingEngine (pipeline/matching/core/engine.py) after the generators/rankers/finalizers for the internal analytics object [. . .]." Unless this was a mistake I did create a class variable; but if this should be an instance variable instance, I'll revise accordingly.

Reply:
Yes, I was not careful with my words on the issue. But now it is moot point because we don't need any instance or class variables!